### DISCUSSION TITLE
Token Vault / Authenticator: QR Code photo not recognized

### DISCUSSION THREAD

**Discussion by ely-see** [standardnotes/forum#2883] [General]
Hi there,

When taking a QR Code photo on iOS 16 and SN v3.43.5 (249), I always get an error that the code couldn't be read.

Whats the error?

Thanks and best regards,
Andreas

**Comment by effieeee** [comment]
Hi Andreas,

I'm sorry to hear that. As a workaround, may I know if you're able to save the QR as a screenshot or image, then choose to upload it to the app when prompted? Otherwise, you may also extract the secret key when you scan the QR code and examine its URL. See the sample below where the key is in a codeblock, right after the **"secret="** portion of the URL:

otpauth://totp/ACME%20Co:john.doe@email.com?**secret=**`HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ`&issuer=ACME%20Co&algorithm=SHA1&digits=6&period=30

**Comment by ely-see** [comment]
Hi effieeee,

Thank you for your answer. Unfortunately, the workaround with the saved photo doesn't work either. I get the same error message. But I can manually enter the secret key. Nevertheless, a fix in the future would be good.

Thanks and best regards,
Andreas

**Comment by jmarz3** [comment]
> 

Hi,
The workaround I have for this is to use a QR scanning app on your phone that will save the image. 

There are so many in the play store with the same name, that I cant specifically state the one I'm using, but it (and others) will take the QR code on screen or if you have a saved image and will reveal the associated URL/otpauth for you, it'll save the image in your photos library if you wish. 

You can use its image and import that into Standard Notes to get Standard Notes to recognize it. 

This is what worked 100% for me. 

Handy thing is you now have both the QR code and the url, so you can save and use whichever you wish if/when needed at a later date.
